frappe/public/js/frappe/views/reports/query_report.js

```diff
@@ -1112,7 +1112,9 @@ frappe.views.QueryReport = class QueryReport extends frappe.views.BaseList {
 	}
 
 	get_filter_values(raise) {
-		const mandatory = this.filters.filter(f => f.df.reqd);
+
+		// check for mandatory property for filters added via UI
+		const mandatory = this.filters.filter(f => (f.df.reqd || f.df.mandatory));
```

Comment:
Mandatory fields should have `reqd` set, we shouldn't have any usage for `mandatory`